ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -69,106 +70,72 @@ class Endpoints(
   import encoder.implicits._
   import json.JsonProtocol._
   import util.ErrorOps._
-  import Uri.Path._
+
+  // Weird signature required, because it looks like Scala
+  // does not see a difference between Future[HttpResponse] & other inner types.
+  // Without it, it will complain that the function has been defined twice.
+  @nowarn("msg=parameter value ev in method toRoute is never used")
+  def toRoute[A <: HttpResponse](res: Future[A])(implicit
+      ev: A =:= HttpResponse
+  ): Route = _ => res map Complete
+  // I think the arguments need to be lazy
```

Comment:
I thought the route dsl is already lazy? Have you tested this?

Reply:
Wrapping my head around how I can test it correctly :smile: Without lazyness the tests did pass too but are they covering scenario? I have my doubts right now (however I did not checked the tests yet).

Reply:
As discussed in slack. Turns out implicit calls ensure that every route after the path is being evaluated lazily (or to be more correct "call-by-name")
![image](https://user-images.githubusercontent.com/25414896/125618901-7c373a0b-8f03-478a-b129-8a8c321e3990.png)
![image](https://user-images.githubusercontent.com/25414896/125618860-31f53dae-ff75-49b8-8f6d-5615d9849ac9.png)